// office-ui-fabric-react/lib-amd/components/DocumentCard/DocumentCardLocation.styles.js
define(["require", "exports", "../../Styling"], function (require, exports, Styling_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.GlobalClassNames = {
        root: 'ms-DocumentCardLocation'
    };
    exports.getStyles = function (props) {
        var theme = props.theme, className = props.className;
        var palette = theme.palette;
        var classNames = Styling_1.getGlobalClassNames(exports.GlobalClassNames, theme);
        return {
            root: [
                classNames.root,
                {
                    fontSize: Styling_1.FontSizes.small,
                    color: palette.neutralPrimary,
                    display: 'block',
                    padding: '8px 16px',
                    position: 'relative',
                    textDecoration: 'none',
                    overflow: 'hidden',
                    textOverflow: 'ellipsis',
                    whiteSpace: 'nowrap',
                    selectors: {
                        ':hover': {
                            color: palette.themePrimary,
                            cursor: 'pointer'
                        }
                    }
                },
                className
            ]
        };
    };
});
